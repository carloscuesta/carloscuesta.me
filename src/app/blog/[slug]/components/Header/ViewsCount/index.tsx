'use client'
import { useEffect } from 'react'
import useSWR from 'swr'

type Props = { slug: string }
async function fetcher<JSON>(
  input: RequestInfo,
  init?: RequestInit
): Promise<JSON> {
  const response = await fetch(input, init)

  return response.json()
}

const ViewsCount = ({ slug }: Props) => {
  const { data } = useSWR<{ views: number }>(`/api/views/${slug}`, fetcher)

  useEffect(() => {
    if (slug) {
      fetch(`/api/views/${slug}`, { method: 'POST' })
    }
  }, [slug])

  return <>{data?.views} views</>
}

export default ViewsCount
