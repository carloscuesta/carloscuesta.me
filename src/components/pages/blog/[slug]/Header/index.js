// @flow
import React from 'react'

import styles from './styles.module.css'

type Props = {
  title: string,
  datePublished: { formatDate: string, value: string }
}

const Header = (props: Props) => (
  <header className={styles.header}>
    <h1 className={styles.title}>{props.title}</h1>

    <time className={styles.time} dateTime={props.datePublished.value}>
      {props.datePublished.formatDate}
    </time>
  </header>
)

export default Header
