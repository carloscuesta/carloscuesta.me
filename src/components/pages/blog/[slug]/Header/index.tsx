import styles from './styles.module.css'

type Props = {
  datePublished: { formatInWords: string, formatDate: string, value: string },
  readingTime: string,
  title: string
}

const Header = (props: Props) => (
  <header className={styles.header}>
    <h1 className={styles.title}>{props.title}</h1>

    <time className={styles.time} dateTime={props.datePublished.value}>
      {props.datePublished.formatDate}
    </time>

    <span className={styles.readingTime}>{props.readingTime}</span>
  </header>
)

export default Header
