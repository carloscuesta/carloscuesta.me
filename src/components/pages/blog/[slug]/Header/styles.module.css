.header {
  padding-bottom: 1.25em;
}

.title {
  margin-bottom: .25em;
}

.time {
  text-transform: uppercase;
  color: var(--textSecondary);
  font-size: 0.85em;
}
