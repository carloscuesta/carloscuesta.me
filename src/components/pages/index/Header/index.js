// @flow
import type { Element } from 'react'
import Link from 'next/link'

import Wrapper from 'src/components/shared/Wrapper'
import styles from './styles.module.css'

const Header = (): Element<'header'> => (
  <header className={styles.container}>
    <Wrapper>
      <h1 className={styles.title}>Hi, I&#39;m <span className={styles.name}>Carlos Cuesta</span></h1>
      <span className={styles.wave}>👋🏼</span>
      <p className={styles.bio}>
        I&#39;m a <b>Front End Engineer</b> from Barcelona.
        I <a target='_blank' rel='noopener noreferrer' href='https://github.com/carloscuesta'>code</a>, <Link href='/blog'><a>write</a></Link> and build stuff on internet.
      </p>
    </Wrapper>
  </header>
)

export default Header
