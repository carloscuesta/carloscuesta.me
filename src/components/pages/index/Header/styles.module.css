.container {
  padding-top: 3rem;
  margin-bottom: 2rem;
}

.title {
  margin: 0;
  font-size: 3.25em;
  display: inline-block;
  padding: 0;
}

.wave {
  animation-duration: 1.2s;
  animation-iteration-count: infinite;
  animation-name: wave;
  display: inline-block;
  transform-origin: 70% 70%;
  font-size: 3rem;
  margin-left: 1rem;
}

.bio {
  font-size: 1.5rem;
  font-weight: 400;
  margin: 0;
  max-width: 456px;
  padding-top: .5rem;
}

@media (max-width: 768px) {
  .container {
    padding-top: 2rem;
  }

  .title {
    font-size: 2.5rem;
    padding-bottom: 1rem;
  }

  .bio {
    font-size: 1.5rem;
  }

  .wave {
    display: none;
  }
}

@keyframes wave {
  0% { transform: rotate(  0.0deg) }
  10% { transform: rotate(-10.0deg) }
  20% { transform: rotate( 12.0deg) }
  30% { transform: rotate(-10.0deg) }
  40% { transform: rotate(  9.0deg) }
  50% { transform: rotate(  0.0deg) }
  100% { transform: rotate(  0.0deg) }
}
