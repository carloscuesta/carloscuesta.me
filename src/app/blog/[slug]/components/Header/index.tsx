import ViewsCount from './ViewsCount'

type Props = {
  datePublished: { formatInWords: string; formatDate: string; value: string }
  readingTime: string
  title: string
  slug: string
}

const Header = (props: Props) => (
  <header className="mb-6 pb-4">
    <h1 className="mb-2 text-4xl font-extrabold">{props.title}</h1>

    <time className="text-sm opacity-70" dateTime={props.datePublished.value}>
      {props.datePublished.formatDate}
    </time>

    <span className='text-sm opacity-70 before:px-1 before:content-["•"]'>
      {props.readingTime}
    </span>

    <span className='text-sm opacity-70 before:px-1 before:content-["•"]'>
      <ViewsCount slug={props.slug} />
    </span>
  </header>
)

export default Header
