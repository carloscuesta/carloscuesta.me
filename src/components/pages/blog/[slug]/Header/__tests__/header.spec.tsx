import renderer from 'react-test-renderer'

import Header from '../index'
import * as stubs from './stubs'

describe('Header', () => {
  it('should match Header component', () => {
    const wrapper = renderer.create(<Header {...stubs.props} />)

    expect(wrapper).toMatchSnapshot()
  })
})
