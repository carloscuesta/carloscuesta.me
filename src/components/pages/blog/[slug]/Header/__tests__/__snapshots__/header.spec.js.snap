// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match Header component 1`] = `
<header
  className="header"
>
  <h1
    className="title"
  >
    Post title
  </h1>
  <time
    className="time"
    dateTime="2019-12-15 10:00"
  >
    15/12/2019
  </time>
</header>
`;
